models/image.py

```diff
@@ -640,20 +656,29 @@ def save(self, filename=None):
                     if not array_name.endswith('.fits'):
                         array_name += '.fits'
                     extensions.append(array_name)
+                    if not single_file:
+                        files_written[array_name] = extpath
 
-            if single_file and not filename.endswith('.fits'):
-                filename += '.fits'
+            if single_file:
+                files_written = files_written['.image.fits']
+                if not self.filepath.endswith('.fits'):
+                    self.filepath += '.fits'
 
         elif format == 'hdf5':
             # TODO: consider writing a more generic utility to save_image_file that handles either fits or hdf5, etc.
             raise RuntimeError("HDF5 format is not yet supported.")
         else:
             raise ValueError(f"Unknown image format: {format}. Use 'fits' or 'hdf5'.")
 
-        self.filepath = filename
-
-        if not single_file:
-            self.filepath_extensions = extensions
+        # Save the file to the archive and update the database record
+        # (as well as self.filepath, self.filepath_extensions, self.md5sum, self.md5sum_extensions)
+        # (From what we did above, it's already in the right place in the local filestore.)
+        if single_file:
+            super().save( files_written, **kwargs )
```

Comment:
Yeah -- if you look at the top of this method, I list overwrite, no_archive, exists_ok, and verify_md5 as things that might be in **kwargs that you might want to think about.  (The defaults are whatever the defaults in FileOnDiskMixin.save are; by default, it will verify the md5 (so it will always at least read an existing file, by default), and will overwrite only if overwrite is true and the md5sum doesn't match.